packages/language-service/ivy/adapters.ts

```diff
@@ -51,6 +51,14 @@ export class LanguageServiceAdapter implements NgCompilerAdapter {
     return this.project.projectService.toCanonicalFileName(fileName);
   }
 
+  /**
+   * Return the real path of a symlink. This method is required in order to
+   * resolve symlinks in node_modules.
+   */
+  realpath(path: string): string {
+    return this.project.realpath?.(path) || path;
```

Comment:
```suggestion
    return this.project.realpath?.(path) ?? path;
```
nit: I think the nullish operator is more appropriate here?